### PR TITLE
[DONT MERGE]Try styled-components

### PR DESCRIPTION
**Pull request by yesmeck**
Try it:

1. git checkout styled-components
1. npm start
1. open http://127.0.0.1:8001/components/checkbox/#components-checkbox-demo-basic

### CODE REVIEW

**Comment by yesmeck** [comment]
Should we convert all less files to js?

**Comment by yesmeck** [comment]
https://www.styled-components.com/docs/advanced#server-side-rendering

File: components/checkbox/Wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export default styled.label.attrs({
+  className: ({ wrapperClass }: any) => wrapperClass
+})`
+  ${({ prefixCls, theme }: any) => css`
+    .${prefixCls!}-wrapper:hover .${prefixCls!}-inner,
+    &:hover .${prefixCls!}-inner,
+    .${prefixCls!}-input:focus + .${prefixCls!}-inner {
+      border-color: ${theme && theme.primaryColor || primaryColor};
+    }
```
Comment (afc163) [inline components/checkbox/Wrapper.tsx]: That is very uncomfortable..

**Comment by afc163** [comment]
It is a huge project for doing this and cannot be compatible for old less-overriding code. I think we should stick to less.

**Comment by afc163** [comment]
The good part is that we don't need less-loader and import css anymore.

File: components/checkbox/Wrapper.tsx (hunk 1 above)
Comment (yesmeck) [reply components/checkbox/Wrapper.tsx]: 这个写法应该还有提升的空间写得更简洁点（input 和 inner 都用 styled-components）。

Comment (yesmeck) [reply components/checkbox/Wrapper.tsx]: > input 和 inner 都用 styled-components

不过这样就得改 rc-components

**Comment by yesmeck** [comment]
related #5656

**Comment by benjycui** [comment]
We should try this from rc to antd, because we must make sure that styled component will not affect antd-mobile and appx first.

**Comment by yesmeck** [comment]
Another [proposal](https://github.com/yesmeck/styled-antd), don't need rewrite components, only need convert all less files to js, but it's too many works, not worth to do it.

**Pull request closed without merging**